main/src/propagator/ipropagator.hpp

```diff
@@ -168,6 +169,26 @@ class Propagator
         output(first, last, simData.chem, writer);
     }
 
+    void logDomainStats(const DomainType& domain, ParticleDataType& simData)
```

Comment:
Since only the hydro (ParticlesData<AccType>)  field of the ParticleDataType is used inside the function, does it make sense to just pass that instead of the entire type? I see it's passed by reference, my comment is more about having a minimal signature (plus I'd like to understand which approach in general you prefer, minimize the signatures or minimize dependencies wrt externally defined types).

Reply:
For free functions I think it's always desirable to minimize the signatures. In this case, the function is a class member, and `DomainType` a class template, so you could argue either way. Because we might add statistics for more than just hydro, I decided to use the class template.